### ISSUE TITLE
DeeplabV3+ training documentation improvement

### ISSUE THREAD

**Issue by thhart** [tensorflow/tensorflow#44350]
###  Description of issue (what needs changing):

Learning rate documentation and training output documentation.

### Clear description

For ML in general the parameter choice is extremely important to have success. You have done incredible work into research of challenging algorithms which help the whole world to improve. However for me as an solution worker it is incredible hard to mine all the details of your ML framework. I am missing fundamental practical documentation about detailed parameter choice involved during training as well as which algorithm to chose. Nowhere is anything said which parameters are a good start to start learning. I am speaking of parameters like weight_decay, base_learning_rate and so on.
When you don't document this all your work is nonsense and not very helpful.
The same applies to training output and interpretation, what does loss mean related to a specific algorithm? Which algorithm to chose why for what purpose and what are upsides and downsides of either?

I know we are dealing with a very dynamic field of science, every now and then there are new algorithms and approaches. However it is very important to have some kind of documentation available which is more than just here is a notebook start it and you will see. Or here is my model, it is working so cool and fast, take it but leave me alone.
Google or whomever it might affect please put some more effort in this middle layer documentation to have your code and inventions used. And let me say this, yes I read it...

**Comment by bitbeast18** [comment]
@thhart Can you please point to the documentation you are talking about?

**Comment by thhart** [comment]
@bitbeast18 As I stated I am missing fundamental information about the training stage and its parameters involved. This is why I can not point to a specific page.

**Comment by bitbeast18** [comment]
First of all, I agree with you that there is room for improvement in documentation (and there will always be) and a lot of work is already happening in this area.

> ### Clear description
> For ML in general the parameter choice is extremely important to have success. You have done incredible work into research of challenging algorithms which help the whole world to improve. However for me as an solution worker it is incredible hard to mine all the details of your ML framework. I am missing fundamental practical documentation about detailed parameter choice involved during training as well as which algorithm to chose. Nowhere is anything said which parameters are a good start to start learning. I am speaking of parameters like weight_decay, base_learning_rate and so on.

Given that the parameter choice is highly task-dependent, it is impossible to document all the cases and scenarios. Despite that TensorFlow offers a sensible set of defaults for almost every hyperparameter, that generally works well. Also, the official website points to some of the (resources)[https://www.tensorflow.org/resources/learn-ml] that you can use to learn more.

You can also check out these tools [Keras Tuner](https://keras-team.github.io/keras-tuner/), [Hyperopt](https://github.com/hyperopt/hyperopt) for hyperparameter tuning and drop questions in the community [pages](https://www.tensorflow.org/community) to get help.

> When you don't document this all your work is nonsense and not very helpful.
> The same applies to training output and interpretation, what does loss mean related to a specific algorithm? Which algorithm to chose why for what purpose and what are upsides and downsides of either?
> 
> I know we are dealing with a very dynamic field of science, every now and then there are new algorithms and approaches. However it is very important to have some kind of documentation available which is more than just here is a notebook start it and you will see. Or here is my model, it is working so cool and fast, take it but leave me alone.
> Google or whomever it might affect please put some more effort in this middle layer documentation to have your code and inventions used. And let me say this, yes I read it...

Also, I don't think that why and how something works is in the scope of the documentation. If you find that the documentation doesn't mention enough details on how to use a particular API then please report it. But it would be really difficult to describe how/why an API works.

If your task is trivial then checkout off-the-shelf models such as [these](https://github.com/tensorflow/models).
